generator/src/generator.ts

```diff
@@ -1669,6 +1669,47 @@ export class Generator {
     if (this.combineSubpaths) map.combineSubpaths();
     return map.toJSON();
   }
+
+  /**
+   *
+   * @param outputDirectory The output directory, resolved relative to process.cwd(), where the import
+   * map and its files are stored on the file system. Defaults to 'dist'
+   * @param importMapFileName The name of the import map file stored to disk, within the outputDirectory. Defaults to
+   * main.importmap
+   * @param hostedBaseUrl a fully qualified or relative string path used as base url for all URLs within the import
+   * If not provided, hostedBaseUrl defaults to './'
+   */
+  async downloadAll(
+    outputDirectory: string = 'deps',
+    importMapFileName: string = 'main.importmap',
+    hostedBaseUrl: string = './'
+  ) {
+    if (!isNode) {
+      throw new JspmError('downloadAll is only available in nodejs');
+    }
+
+    const fs = await import('node:fs/promises');
+    const path = await import('node:path');
+
+    for (const url in this.traceMap.installer.resolver.traceEntries) {
```

Comment:
I'm not sure if this captures all of the files that could possibly be used. I'm not sure if it's possible for jspm-generator import maps to load files that then load additional modules that are not contained within the import map.

I have [another implementation of this](https://github.com/single-spa/import-map-microfrontend-deps/blob/62a9f6cc6c6a93e8752c94d5c6b5cbd6bfd9ee09/src/import-map-microfrontend-deps.ts#L101) within the single-spa ecosystem, and used the `integrity` field of the import map to obtain the urls to download, rather than the `resolver` within jspm generator.

This is my first contribution to the project - looking forward to feedback on it.